columnar/src/dynamic_column.rs

```diff
@@ -20,6 +22,114 @@ pub enum DynamicColumn {
     Str(StrColumn),
 }
 
+impl DynamicColumn {
+    pub fn is_numerical(&self) -> bool {
+        self.is_u64() || self.is_i64() || self.is_f64()
+    }
+    pub fn is_f64(&self) -> bool {
+        matches!(self, DynamicColumn::F64(_))
+    }
+    pub fn is_u64(&self) -> bool {
+        matches!(self, DynamicColumn::U64(_))
+    }
+    pub fn is_i64(&self) -> bool {
+        matches!(self, DynamicColumn::I64(_))
+    }
+
+    pub fn coerce_to_f64(self) -> Option<DynamicColumn> {
+        match self {
+            DynamicColumn::I64(column) => Some(DynamicColumn::F64(Column {
+                idx: column.idx,
+                values: Arc::new(monotonic_map_column(column.values, MapI64ToF64)),
+            })),
+            DynamicColumn::U64(column) => Some(DynamicColumn::F64(Column {
+                idx: column.idx,
+                values: Arc::new(monotonic_map_column(column.values, MapU64ToF64)),
+            })),
+            DynamicColumn::F64(_) => Some(self),
+            _ => None,
+        }
+    }
+    pub fn coerce_to_i64(self) -> Option<DynamicColumn> {
+        match self {
+            DynamicColumn::U64(column) => {
+                if column.max_value() > i64::MAX as u64 {
+                    return None;
+                }
+                Some(DynamicColumn::I64(Column {
+                    idx: column.idx,
+                    values: Arc::new(monotonic_map_column(column.values, MapU64ToI64)),
+                }))
+            }
+            DynamicColumn::I64(_) => Some(self),
+            _ => None,
+        }
+    }
+    pub fn coerce_to_u64(self) -> Option<DynamicColumn> {
+        match self {
+            DynamicColumn::I64(column) => {
+                if column.min_value() < 0 {
+                    return None;
+                }
+                Some(DynamicColumn::U64(Column {
+                    idx: column.idx,
+                    values: Arc::new(monotonic_map_column(column.values, MapI64ToU64)),
+                }))
+            }
+            DynamicColumn::U64(_) => Some(self),
+            _ => None,
+        }
+    }
+}
+
+pub struct MapI64ToF64;
```

Comment:
Do you think we could replace those three thing by the following?
I don;'t know any trait boudnary for as.

If you want to stick to `as`, can you make those private to this module?
Technically these mapping are not monotonic.



```
pub struct Coerce<FromTyp, ToTyp>;

impl<FromTyp, ToTyp> StrictlyMonotonicFn<FromTyp, ToTyp> for MapI64ToF64 
where FromTyp: TryFrom<ToTyp>, ToTyp: TryFrom<FromTyp>
{
}
```

Reply:
maybe with num traits, but the use case is too small

Reply:
ok